mqtt/mqtt-broker/src/session/mod.rs

```diff
@@ -27,17 +26,15 @@ pub enum Session {
 }
 
 impl Session {
-    pub fn new_transient(auth_id: AuthId, connreq: ConnReq, state: SessionState) -> Self {
-        let (client_id, peer_addr, connect, handle) = connreq.into_parts();
-        let client_info = ClientInfo::new(client_id, peer_addr, auth_id);
-        let connected = ConnectedSession::new(state, client_info, connect.will, handle);
+    pub fn new_transient(connreq: ConnReq, state: SessionState) -> Self {
```

Comment:
nit: I wonder if we can just pass will message into both methods instead of ConnReq

Reply:
if you do that you may not need to keep getters for `client_id` and `peer_addr` in the ConnReq

Reply:
I suggested to @dylanbronson to create another type `SessionReq` and convert ConnReq into SessionReq. That can help simplify code and types, but we agreed to try that later in a separate PR.

Reply:
Just curious what will be the purpose of `SessionReq`?